sim/firesim-lib/src/main/cc/fesvr/firesim_fesvr.cc

```diff
@@ -1,175 +1,74 @@
 //See LICENSE for license details
 #include "firesim_fesvr.h"
 
-#define MSIP_BASE 0x2000000
 
-#define NHARTS_MAX 16
-
-int firesim_fesvr_t::host_thread(void *arg)
-{
-    firesim_fesvr_t *fesvr = static_cast<firesim_fesvr_t*>(arg);
-    fesvr->run();
-
-    while (true)
-        fesvr->target->switch_to();
-
-    return 0;
-}
-
-firesim_fesvr_t::firesim_fesvr_t(const std::vector<std::string> args, bool has_loadmem) : htif_t(args), has_loadmem(has_loadmem)
+firesim_fesvr_t::firesim_fesvr_t(int argc, char** argv, bool can_have_loadmem) : testchip_fesvr_t(argc, argv, has_loadmem), is_busy(false)
```

Comment:
This is a strange style where some of the variable inits are up here but `has_loadmem(can_have_loadmem)` isn't.

Reply:
`has_loadmem` is a property of the parent `testchip_fesvr_t` class, so it can't be added to the initialization list here.

For SW sim, `loadmem` is only enabled if the `+loadmem` flag is seen, so `testchip_fesvr_t` only sets `has_loadmem` if `can_have_loadmem` and `+loadmem` is present.

For FireSim, there is no `+loadmem` flag, so we have to override the behavior in `testchip_fesvr_t` and always set `has_loadmem` if `can_have_loadmem`